src/sys/socket/mod.rs

```diff
@@ -173,6 +170,47 @@ libc_bitflags!{
     }
 }
 
+/// Unix credentials of the sending process.
+///
+/// This struct is used with the `SCM_CREDENTIALS` ancillary message for UNIX sockets.
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct UnixCredentials(libc::ucred);
```

Comment:
should be `libc::xucred` on FreeBSD.

Reply:
This is a very different looking struct, I'm really surprised that this is correct. I would have thought it'd be the struct I mentioned above, `cmsgcred`, which does have pid, uid, and gid fields.

Reply:
Except there are two problems.  If `UnixCredentials` were actually used with `SCM_CREDENTIALS`, then you would be correct.  `struct cmsgcred` would be the struct to wrap.  However, the `SCM_CREDENTIALS` comment is red herring.  `UnixCredentials` is actually used as a socket option, with `SO_PEERCRED`.  That's why I say that it should wrap `xucred`.

Reply:
From [this SO question](https://stackoverflow.com/questions/10037086/so-peercred-vs-scm-credentials-why-there-are-both-of-them#11213530) it seems like `SCM_CREDENTIALS` should be what we offer as an API and we abandon the `SO_PEERCRED` implementation, since one looks to be a newer/better version of the other.